src/include/plan_node/create_plan_node.h

```diff
@@ -0,0 +1,970 @@
+      trigger_name_ = std::move(trigger_name);
+      return *this;
+    }
+
+    /**
+     * @param trigger_funcnames trigger function names for [CREATE TRIGGER]
+     * @return builder object
+     */
+    Builder &SetTriggerFuncnames(std::vector<std::string> &&trigger_funcnames) {
+      trigger_funcnames_ = std::move(trigger_funcnames);
+      return *this;
+    }
+
+    /**
+     * @param trigger_args trigger args for [CREATE TRIGGER]
+     * @return builder object
+     */
+    Builder &SetTriggerArgs(std::vector<std::string> &&trigger_args) {
+      trigger_args_ = std::move(trigger_args);
+      return *this;
+    }
+
+    /**
+     * @param trigger_columns trigger columns for [CREATE TRIGGER]
+     * @return builder object
+     */
+    Builder &SetTriggerColumns(std::vector<std::string> &&trigger_columns) {
+      trigger_columns_ = std::move(trigger_columns);
+      return *this;
+    }
+
+    /**
+     * @param trigger_when trigger when clause for [CREATE TRIGGER]
+     * @return builder object
+     */
+    Builder &SetTriggerWhen(std::shared_ptr<parser::AbstractExpression> trigger_when) {
+      trigger_when_ = std::move(trigger_when);
+      return *this;
+    }
+
+    /**
+     * @param trigger_type trigger type, i.e. information about row, timing, events, access by pg_trigger
+     * @return builder object
+     */
+    Builder &SetTriggerType(int16_t trigger_type) {
+      trigger_type_ = trigger_type;
+      return *this;
+    }
+
+    /**
+     * @param view_name  view name for [CREATE VIEW]
+     * @return builder object
+     */
+    Builder &SetViewName(std::string view_name) {
+      view_name_ = std::move(view_name);
+      return *this;
+    }
+
+    /**
+     * @param view_query view query for [CREATE VIEW]
+     * @return builder object
+     */
+    Builder &SetViewQuery(std::shared_ptr<parser::SelectStatement> view_query) {
+      view_query_ = std::move(view_query);
+      return *this;
+    }
+
+    /**
+     * @param create_stmt the SQL CREATE statement
+     * @return builder object
+     */
+    Builder &SetFromCreateStatement(parser::CreateStatement *create_stmt) {
+      switch (create_stmt->GetCreateType()) {
+        case parser::CreateStatement::CreateType::kDatabase: {
+          create_type_ = CreateType::DB;
+          database_name_ = std::string(create_stmt->GetDatabaseName());
+          break;
+        }
+
+        case parser::CreateStatement::CreateType::kSchema: {
+          create_type_ = CreateType::SCHEMA;
+          database_name_ = std::string(create_stmt->GetDatabaseName());
+          schema_name_ = std::string(create_stmt->GetSchemaName());
+          break;
+        }
+
+        case parser::CreateStatement::CreateType::kTable: {
+          table_name_ = std::string(create_stmt->GetTableName());
+          schema_name_ = std::string(create_stmt->GetSchemaName());
+          database_name_ = std::string(create_stmt->GetDatabaseName());
```

Comment:
1. General comment on create plan node. Big. Amazingly big. Was consideration given to breaking it up into create sub-types, and if so, what prevented that approach?

2. GetDatabaseName(). This is inconsistent across plan node types.

Somewhere associated with each connection, we will have a database for that connection. We can store the database name, but in addition may wish to store the oid, since the latter is more directly useful. Therefore, in plan nodes, choices:

1) Don't store database at all, retrieve from connection.
2) Store database (consistently) across plan node types. Name possibly, oid preferred, or both. (Except for create database, where it has to be name).